### ISSUE TITLE
Gba port?

### ISSUE THREAD

**Issue by InfamousKnight** [SquirrelJME/SquirrelJME#91]
Would it be possible to port squirreljme to gba rom? Mainly to benefit bizhawk users.

**Comment by XerTheSquirrel** [comment]
I know of the TAS aspect of BizHawk, but it is possible with the RatufaCoat VM that is going into RetroArch as it will be having save states and such.

As for a port to GBA, it has 256KB of RAM. With a cart size of 32MiB it could fit SquirrelJME just fine, but the major issue is RAM so it would probably have to be a special cart with extra memory.

**Comment by XerTheSquirrel** [comment]
Will definitely be trying the Original DS for a port.

A GBA port might be possible if:

 * A flash cart with "ROM" that can be used as RAM.
 * The ROM address space can be written to and read from if so.

But this would require a specialized cart and I am making assumptions about how the GBA works.

**Comment by endrift** [comment]
Do you need any of these:

- Memory mapping
- Memory protection
- Data or prefetch aborts at all
- A framebuffer that can be updated in full at 60 fps

If so, then the GBA can't do that, even with a special RAM cartridge, which would need to be specially made for this anyway.

Also, I'm really curious how you could run this on a platform that, uh, doesn't have a JVM ported to it.

**Comment by XerTheSquirrel** [comment]
> Do you need any of these:
>     * Memory mapping
>     * Memory protection
>     * Data or prefetch aborts at all
>     * A framebuffer that can be updated in full at 60 fps

None of these are needed at all.

> Also, I'm really curious how you could run this on a platform that, uh, doesn't have a JVM ported to it.

SquirrelJME is the JVM itself, it is completely standalone.

**Comment by endrift** [comment]
Ah, I misunderstood the scope of this then. I found rakuta only after I posted that but I understand now.

The real problems then are the fact that the GBA is...pretty dang slow, and also the lack of RAM. Especially fast RAM. And there isn't really a good solution for that that doesn't involve emulator or hardware hacks made specifically for this, which is not something anyone really wants to do I'm sure.
